### ISSUE TITLE
Options Timeout whats meaning

### ISSUE THREAD

**Issue by mnd5756242** [projectdiscovery/nuclei#5206]
type Options struct {
Timeout int
}
The Timeout in this options represents the global request time or the http request time of each template.

I set Timeout like this nuclei v3 version
options := types.DefaultOptions()
options.Timeout = 18

**Comment by GeorginaReeder** [comment]
Thanks for your submission @mnd5756242 , we'll take a look into it! :)

**Comment by Mzack9999** [comment]
@mnd5756242 it's the global timeout that the engine will attempt to request for each protocol. You can override it within http via request annotation:

```
@Timeout: xxx
GET / HTTP/1.1
Host: host.here
```

**Comment by chushuai** [comment]
@Mzack9999  In other words, the priority of @Timeout is higher than that of options.Timeout.

**Comment by Mzack9999** [comment]
Correct, for http protocol. I'm closing the issue as it seems the question has been answered. Feel free to reopen if needed.
